Add AGS10 Sensor (#3527)
OttoWinter/esphomedocs: components/sensor/ags10.rst

AGS10 Volatile Organic Compound (VOC) Sensor
============================================

.. seo::
    :description: Instructions for setting up AGS10 VOC sensors with ESPHome
    :image: ags10.jpg
    :keywords: AGS10

The ``ags10`` sensor platform VOC sensor allows you to use your ASAIR AGS10
(`datasheet <http://www.aosong.com/userfiles/files/Datasheet%20AGS10.pdf>`__,
`ASAIR`_ ) sensors with
ESPHome. The :ref:`I²C Bus <i2c>` is
required to be set up in your configuration for this sensor to work.

.. warning::

    The sensor supports up to 15kHz operation, so you should specify up to ``frequency: 15kHz`` in your ``i2c`` configuration.

.. _ASAIR: http://www.aosong.com/en/products-86.html

.. figure:: images/ags10.jpg
    :align: center
    :width: 30.0%

    AGS10 VOC Sensor

.. code-block:: yaml

    # Example configuration entry
    sensor:
      - platform: ags10
        tvoc:
          name: TVOC

Configuration variables:
------------------------

- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.

>   All other options from :ref:`Sensor <config-sensor>`.

Reviewer: ```suggestion
  All options from :ref:`Sensor <config-sensor>`.
```

- **address** (*Optional*, int): Manually specify the I²C address of
  the sensor. Defaults to ``0x1A``.
- **update_interval** (*Optional*, :ref:`config-time`): The interval to check the
  sensor. Defaults to ``60s``.
- **version** (*Optional*, int): The firmware version of the sensor.

> - **resistance** (*Optional*, int): The initial value of the sesor resistance.

Reviewer: ```suggestion
- **resistance** (*Optional*, int): The initial value of the sensor resistance.
```

> - **version** (*Optional*, int): The firmware version of the sensor.
> - **resistance** (*Optional*, int): The initial value of the sensor resistance.

Reviewer: ```suggestion
- **version** (*Optional*): The firmware version of the sensor.
  All options from :ref:`Sensor <config-sensor>`.
- **resistance** (*Optional*): The initial value of the sensor resistance.
  All options from :ref:`Sensor <config-sensor>`.
```

AGS10 Volatile Organic Compound (VOC) Sensor
============================================

.. seo::
    :description: Instructions for setting up AGS10 VOC sensors with ESPHome
    :image: ags10.jpg
    :keywords: AGS10

The ``ags10`` sensor platform VOC sensor allows you to use your ASAIR AGS10
(`datasheet <http://www.aosong.com/userfiles/files/Datasheet%20AGS10.pdf>`__,
`ASAIR`_ ) sensors with
ESPHome. The :ref:`I²C Bus <i2c>` is
required to be set up in your configuration for this sensor to work.

> .. warning::

Reviewer: A warning is a bit scary for this
```suggestion
.. note::
```

AGS10 Volatile Organic Compound (VOC) Sensor
============================================

.. seo::
    :description: Instructions for setting up AGS10 VOC sensors with ESPHome
    :image: ags10.jpg
    :keywords: AGS10

The ``ags10`` sensor platform VOC sensor allows you to use your ASAIR AGS10
(`datasheet <http://www.aosong.com/userfiles/files/Datasheet%20AGS10.pdf>`__,
`ASAIR`_ ) sensors with
ESPHome. The :ref:`I²C Bus <i2c>` is
required to be set up in your configuration for this sensor to work.

.. warning::

    The sensor supports up to 15kHz operation, so you should specify up to ``frequency: 15kHz`` in your ``i2c`` configuration.

.. _ASAIR: http://www.aosong.com/en/products-86.html

.. figure:: images/ags10.jpg
    :align: center
    :width: 30.0%

    AGS10 VOC Sensor

.. code-block:: yaml

>     # I²C Bus configuration
>     i2c:
>       sda: GPIO21
>       scl: GPIO22
>       scan: true
>       frequency: 15kHz
> 
>     # Example configuration entry
>     sensor:
>       - platform: ags10
>         tvoc:
>           name: TVOC
>         address: 0x1A
>         update_interval: 60s

Reviewer: Example YAML should be basic to allow the user to read the documentation and understand how to configure it
```suggestion
    # Example configuration entry
    sensor:
      - platform: ags10
        tvoc:
          name: TVOC
```

Author: Done

> Configuration variables:

Reviewer: `address` and `resistance` sensors are missing from the documentation.

Author: `version` and `resistance`. The `address` was already described.
Done.

------------------------

- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.

  - **name** (**Required**, string): The name for the tvoc sensor.
  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.

>   - All other options from :ref:`Sensor <config-sensor>`.

Reviewer: We don't duplicate config lines that are linked in sensor core docs
```suggestion
- **tvoc** (**Required**): The information for the total Volatile Organic Compounds sensor.
  All options from :ref:`Sensor <config-sensor>`.
```

Author: Done.
